oidc/oidc.go

```diff
@@ -201,35 +203,53 @@ func (p *ProviderConfig) NewProvider(ctx context.Context) *Provider {
 	}
 }
 
+var oidcConfigCache, _ = ristretto.NewCache(&ristretto.Config[string, []byte]{
+	NumCounters: 10_000,
+	MaxCost:     10_000 * 1024 * 1024, // 10k items, 1MB each
+	BufferItems: 64,
```

Comment:
1. Counters is recommended to be 10 * max cost
2. max cost is the maximum cost of all items. You define cost as `len(body)` below - so the real cost function is `len(key) + len(body)`. In this case, one cost item = 1 byte, so this is 10gb of cache

Reply:
1. Counters is not recommended to be 10 * max cost but 10 * max number of items (keys).
    > For example, if you expect your cache to hold 1,000,000 items when full, NumCounters should be 10,000,000 (10x).
2. Ok true, this is too big. How many items do we expect the cache to hold, and how big would they be? Probably 1MB per item is too generous already.

Reply:
1. I stand corrected!
2. We probably have a few hundred items in cache at most because for most social sign in providers the config is the same. This is really only for sso? So maybe ~20 for all social providers and then ~100-200 for frequently used sso providers? Well known endpoints are probably in the 2-5kb range